contentcuration/contentcuration/frontend/shared/vuex/connectionCheckerPlugin/connectionModule.js

```diff
@@ -0,0 +1,74 @@
+import axios from 'axios';
+
+export default {
+  state: () => ({
+    online: true,
+    timeLastWentOffline: null,
+    timeLastCameOnline: null,
+    polling: false,
+  }),
+  getters: {
+    online(state) {
+      return state.online && navigator.onLine;
+    },
+  },
+  mutations: {
+    SET_ONLINE_STATUS(state, online) {
+      if (state.online != online) {
+        // the state has changed
+        state.timeLastWentOffline = !online ? new Date() : null;
+        state.timeLastCameOnline = online ? new Date() : null;
+        state.online = online;
+      }
+    },
+    START_POLLING(state) {
+      state.polling = true;
+    },
+    STOP_POLLING(state) {
+      state.polling = false;
+    },
+  },
+  actions: {
+    handleBrokenConnection({ commit, dispatch }) {
+      commit('SET_ONLINE_STATUS', false);
+      dispatch('checkConnection');
+    },
+    handleReconnection({ commit, state }) {
+      if (!state.online) commit('SET_ONLINE_STATUS', true);
+      if (state.polling) commit('STOP_POLLING');
+    },
+    checkConnection({ state, commit, dispatch }) {
+      // used https://github.com/Aupajo/backoff-calculator to tune this
+      let maximumPollingDelay = 30 * 60; // 30 minutes
+      let initialPollingDelay = 1; // 1 second
+      let delaySeconds = i => Math.min(i ** 2 + initialPollingDelay, maximumPollingDelay);
+
+      if (state.polling) {
+        return; // polling has already been initiated
+      } else {
+        commit('START_POLLING');
+      }
```

Comment:
Seems like this `if/else` block could be moved above the variables, since they wouldn't be used if `state.polling` is `true`.

Reply:
I ended up splitting this up.  The if statement is now at the top of the function, and to make sure that the `START_POLLING` action doesn't happen unless the first polling request is made successfully, I put `commit('START_POLLING')` at the very end of the function.